src/rdkafka_telemetry_encode.c

```diff
@@ -258,20 +257,131 @@ static const rd_kafka_telemetry_metric_value_calculator_t
+
+static const char *get_member_id(const rd_kafka_t *rk) {
+        return rk->rk_cgrp->rkcg_member_id &&
+                       rk->rk_cgrp->rkcg_member_id->len > 0
+                   ? (const char *)rk->rk_cgrp->rkcg_member_id->str
+                   : NULL;
+}
+
+static const char *get_transactional_id(const rd_kafka_t *rk) {
+        return rk->rk_conf.eos.transactional_id
+                   ? (const char *)rk->rk_conf.eos.transactional_id
+                   : NULL;
+}
+
+static const rd_kafka_telemetry_attribute_config_t producer_attributes[] = {
+    {"client_rack", get_client_rack},
+    {"transactional_id", get_transactional_id},
+};
+
+static const rd_kafka_telemetry_attribute_config_t consumer_attributes[] = {
+    {"client_rack", get_client_rack},
+    {"group_id", get_group_id},
+    {"group_instance_id", get_group_instance_id},
+    {"member_id", get_member_id},
+};
+
+static int
+count_attributes(rd_kafka_t *rk,
+                 const rd_kafka_telemetry_attribute_config_t *configs,
+                 int config_count) {
+        int count = 0, i;
+        for (i = 0; i < config_count; ++i) {
+                if (configs[i].getValue(rk)) {
+                        count++;
+                }
+        }
+        return count;
+}
+
+static void set_attributes(rd_kafka_t *rk,
+                           rd_kafka_telemetry_resource_attribute_t **attributes,
+                           const rd_kafka_telemetry_attribute_config_t *configs,
+                           int config_count) {
+        int attr_idx = 0, i;
+        for (i = 0; i < config_count; ++i) {
+                const char *value = configs[i].getValue(rk);
+                if (value) {
+                        attributes[attr_idx]->name  = configs[i].name;
+                        attributes[attr_idx]->value = value;
+                        attr_idx++;
+                }
+        }
+}
+
+static int
+resource_attributes(rd_kafka_t *rk,
+                    rd_kafka_telemetry_resource_attribute_t **attributes) {
+        int count = 0;
+        const rd_kafka_telemetry_attribute_config_t *configs;
+        int config_count;
+
+        if (rk->rk_type == RD_KAFKA_PRODUCER) {
+                configs      = producer_attributes;
+                config_count = sizeof(producer_attributes) /
+                               sizeof(producer_attributes[0]);
+        } else if (rk->rk_type == RD_KAFKA_CONSUMER) {
+                configs      = consumer_attributes;
+                config_count = sizeof(consumer_attributes) /
+                               sizeof(consumer_attributes[0]);
+        } else {
+                *attributes = NULL;
+                return 0;
+        }
+
+        count = count_attributes(rk, configs, config_count);
+
+        if (count == 0) {
+                *attributes = NULL;
+                return 0;
+        }
+
+        *attributes =
+            rd_malloc(sizeof(rd_kafka_telemetry_resource_attribute_t) * count);
+        if (!*attributes) {
```

Comment:
no need to handle this, rd_malloc will just crash us if the memory alloc fails

Reply:
Makes sense, removed